packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -180,22 +188,16 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         });
 
         commands.registerCommand(TerminalCommands.TERMINAL_CONTEXT, new UriAwareCommandHandler<URI>(this.selectionService, {
-            execute: async uri => {
-                // Determine folder path of URI
-                const stat = await this.fileSystem.getFileStat(uri.toString());
-                if (!stat) {
-                    return;
+            execute: async uri => this.openTerminalAtUri(uri)
+        }));
+        commands.registerCommand(TerminalCommands.TERMINAL_CONTEXT_ROOT, {
+            execute: async () => {
+                if (this.workspaceService.workspace) {
```

Comment:
i don't think using `this.workspaceService.workspace` is correct generally, since it can be just a file. We should base command on the current selection and roots, not a workspace.